public/pages/MonitorDetails/containers/MonitorDetails.js

```diff
@@ -174,14 +178,189 @@ export default class MonitorDetails extends Component {
 
-  onEditTrigger = trigger => {
+  onEditTrigger = (trigger) => {
     this.setState({ triggerToEdit: trigger });
     this.props.history.push({
       ...this.props.location,
       search: `?action=${TRIGGER_ACTIONS.UPDATE_TRIGGER}`,
     });
   };
 
+  convertToADConfigs = async (monitor) => {
+    const uiMetadata = _.get(monitor, 'ui_metadata');
+    let adName = monitor.name + '-Detector';
+    let adTimeField = uiMetadata.search.timeField;
+    let adIndices = monitor.inputs[0].search.indices;
+    let adDetectorInterval = { period: { interval: 1, unit: 'MINUTES' } };
+    const {
+      frequency,
+      period: { interval, unit },
+      daily,
+      weekly,
+      monthly: { day },
+      cronExpression,
+      timezone,
+    } = _.get(uiMetadata, 'schedule', {});
+    const search = _.get(uiMetadata, 'search');
+    if (frequency === 'interval') {
+      adDetectorInterval = { period: { unit: unit, interval: interval } };
+    }
+    let windowDelay = await this.getLatestTimeStamp(adTimeField, adIndices);
+    let adWindowDelay = { period: { interval: 10, unit: 'MINUTES' } };
+    if (adWindowDelay) {
+      adWindowDelay = { period: { interval: windowDelay, unit: 'MINUTES' } };
+    }
+    let filterQuery = formikToWhereClause(search);
+    let adFilterQuery = '';
+    if (filterQuery) {
+      adFilterQuery = { bool: { filter: [filterQuery] } };
+    } else {
+      adFilterQuery = { match_all: { boost: 1.0 } };
+    }
+    let { aggregationType, fieldName } = search;
+    let adFeatures;
+    if (!aggregationType || !fieldName) {
+      adFeatures = {};
+    } else {
+      if (aggregationType == 'count') {
+        aggregationType = 'value_count';
+      }
+      adFeatures = {
+        feature_name: 'feature-1',
+        feature_enabled: true,
+        aggregation_query: { aggregation_name: { [aggregationType]: { field: fieldName } } },
+      };
+    }
+    let adConfigs = {
+      name: adName,
+      description: '',
+      time_field: adTimeField,
+      indices: adIndices,
+      feature_attributes: [adFeatures],
+      filter_query: adFilterQuery,
+      detection_interval: adDetectorInterval,
+      window_delay: adWindowDelay,
+    };
+    let queriesForOverview = {
+      filter_query: displayText(_.get(search, 'where')),
+      feature_attributes: {
+        feature_name: 'feature-1',
+        aggregationType: aggregationType,
+        fieldName: fieldName,
+      },
+      where: _.get(search, 'where'),
+    };
+    let validationResponse = await this.validateADConfigs(adConfigs);
+    this.decideWhichFlyout(validationResponse, adConfigs, queriesForOverview);
+  };
+
+  decideWhichFlyout = (validationResponse, adConfigs, queriesForOverview) => {
+    const setFlyout = this.props.setFlyout;
+    if (validationResponse.failures.others) {
+      let message = '';
+      for (let [key, value] of Object.entries(validationResponse.failures)) {
+        if (key === 'others') {
+          message = value;
+        }
+      }
```

Comment:
Can there be multiple failures with a key of `others`? Also, confused on what `others` means here, could you explain?